include/filters/filter_base.hpp

```diff
@@ -110,26 +110,44 @@ class FilterBase
    */
   virtual bool configure()=0;
 
+private:
+  template <typename PT>
+  bool getParamImpl(const std::string & name, const uint8_t type, PT default_value, PT & value_out)
+  {
+    std::string param_name = param_prefix_ + name;
+
+    if (!params_interface_->has_parameter(param_name)) {
+      // Declare parameter
+      rclcpp::ParameterValue default_parameter_value(default_value);
+      rcl_interfaces::msg::ParameterDescriptor desc;
+      desc.name = name;
+      desc.type = type;
+      desc.read_only = true;
+
+      if (name.size() == 0u) {
+        throw std::runtime_error("Parameter must have a name");
+      }
+
+      params_interface_->declare_parameter(
+          param_name,
+          default_parameter_value,
+          desc);
+    }
+
+    value_out = params_interface_->get_parameter(param_name).get_parameter_value().get<PT>();
+    // TODO(sloretz) seems to be no way to tell if parameter was initialized or not
+    return true;
+  }
+
+protected:
```

Comment:
Can we collapse all of the protected sections into one?  And then put the private section after that?